Modules/Challenges/src/Challenges.Infrastructure/Persistence/IServiceCollectionExtensions.cs

```diff
@@ -8,23 +7,41 @@ namespace Backbone.Modules.Challenges.Infrastructure.Persistence;
 
 public static class IServiceCollectionExtensions
 {
+    private const string SQLSERVER = "SqlServer";
+    private const string SQLSERVER_MIGRATIONS_ASSEMBLY = "Challenges.Infrastructure.Database.SqlServer";
+    private const string POSTGRES = "Postgres";
+    private const string POSTGRES_MIGRATIONS_ASSEMBLY = "Challenges.Infrastructure.Database.Postgres";
+
     public static void AddDatabase(this IServiceCollection services, Action<DbOptions> setupOptions)
     {
         var options = new DbOptions();
         setupOptions?.Invoke(options);
 
-        services.AddDbContext<ApplicationDbContext>(dbContextOptions =>
-            dbContextOptions.UseSqlServer(options.DbConnectionString, sqlOptions =>
+        services.AddDbContext<ChallengesDbContext>(dbContextOptions => _ = options.Provider switch
```

Comment:
Wouldn't the following be more readable?
```cs
switch(options.Provider) {
    case SQLSERVER: 
        services.AddDbContext<ChallengesDbContext>(dbContextOptions => 
            dbContextOptions.UseSqlServer(options.DbConnectionString, sqlOptions =>
            {
                sqlOptions.CommandTimeout(20);
                sqlOptions.MigrationsAssembly(SQLSERVER_MIGRATIONS_ASSEMBLY);
                sqlOptions.EnableRetryOnFailure(options.RetryOptions.MaxRetryCount, TimeSpan.FromSeconds(options.RetryOptions.MaxRetryDelayInSeconds), null);
            })
        );
        break;
    case SQLSERVER: 
        services.AddDbContext<ChallengesDbContext>(dbContextOptions => 
            dbContextOptions.UseNpgsql(options.DbConnectionString, sqlOptions =>
            {
                sqlOptions.CommandTimeout(20);
                sqlOptions.MigrationsAssembly(POSTGRES_MIGRATIONS_ASSEMBLY);
                sqlOptions.EnableRetryOnFailure(options.RetryOptions.MaxRetryCount, TimeSpan.FromSeconds(options.RetryOptions.MaxRetryDelayInSeconds), null);
            }),
        );
        break;
}
```

(probably not valid C#, but you get the idea. ;-)